main/src/ca/uwaterloo/flix/api/lsp/provider/completion/CompletionUtils.scala

```diff
@@ -228,4 +229,33 @@ object CompletionUtils {
   private def splitByCamelCase(input: String): List[String] = {
     input.split("(?=[A-Z])").toList
   }
+
+  /**
+    * Parse given `qn` based on the character immediately following the location.
+    * If the character is a dot, we move the ident to the namespace.
+    * Otherwise, we will just take namespace and ident out of qn.
+    *
+    * Example:
+    *   - Source "A.B.C", QName(["A", "B"], "C") -> ("A.B", "C")
+    *   - Source "A.B.C.", QName(["A", "B"], "C") -> ("A.B.C", "")
+    */
+  def getNamespaceAndIdentFromQName(qn: QName, loc: SourceLocation): (List[String], String) = {
+    val ident = if (followedByDot(loc)) "" else qn.ident.name
+    val namespace = qn.namespace.idents.map(_.name) ++ {
+      if (followedByDot(loc)) List(qn.ident.name)
+      else Nil
+    }
+    (namespace, ident)
+  }
+
+  /**
+    * Returns true if the character immediately following the location is a dot.
+    * Note:
+    *   - loc.endCol will point to the next character after QName.
+    *   - loc is 1-indexed, so we are actually checking the character at loc.endCol-1.
+    */
+  private def followedByDot(loc: SourceLocation): Boolean = {
```

Comment:
Change this as follows:
- Instead of loc, we take the qname of the error.
- We then check if qname.loc.column + 1 is a dot (we have to check its within bounds)